test/test_encoding.py

```diff
@@ -0,0 +1,743 @@
+#!/usr/bin/python3 -u
+"""
+SPDX-License-Identifier: BSD-2
+"""
+import unittest
+
+from tpm2_pytss import *
+from tpm2_pytss.encoding import (
+    base_encdec,
+    toolslike_encdec,
```

Comment:
Is it supposed to be tools like or actually match the tools? If it should match the tools, lets not be ambiguous about that and call it tpm2tools_encdec.

Plus pep8 classes should be camelCased.

Reply:
Whats the difference, ordering or missing fields?

Reply:
different field names and tpm2-tools flatten some structures, for example tpm2_readpublic puts the RSA parameters at the same level as type and nameAlg, so toolslike_encdec should be seen more as TPMA_FRIENDLY_INTLIST.parse then a compatibility layer for the tools

Reply:
> different field names and tpm2-tools flatten some structures, for example tpm2_readpublic puts the RSA parameters at the same level as type and nameAlg, so toolslike_encdec should be seen more as TPMA_FRIENDLY_INTLIST.parse then a compatibility layer for the tools

Folks over in tpm.dev are clamoring for a way to get tools YAML in and out to match. Is their anyway we can add in the special handling for tools?

Reply:
It's possible, I wouldn't do that in this class tho
It's just a lot of work going through all the source code for the tools and implementing + verifying it

Reply:
> It's possible, I wouldn't do that in this class tho It's just a lot of work going through all the source code for the tools and implementing + verifying it

Let's take base and json now, and then start a longer running branch for tools.

Reply:
FYI the tools YAML support is indicated in ticket: https://github.com/tpm2-software/tpm2-pytss/issues/235

Reply:
> 
> Let's take base and json now, and then start a longer running branch for tools.

I would like to keep toolslike as well, as I have at least two use cases for that
I'll see if I can create a new branch of this and start the tools-part there in a couple of days

Reply:
> > Let's take base and json now, and then start a longer running branch for tools.
> 
> I would like to keep toolslike as well, as I have at least two use cases for that I'll see if I can create a new branch of this and start the tools-part there in a couple of days

The problem with tools-like, is that if we merge it and don't make it tools-firm by release, we have effectively made a new format. We already have enough formats to deal with. If you want to hide it behind a feature flag where to use it you have to buy into the fact it's experimental, that's fine too. You could just make it _toolslike_encdec indicating that it's private still and you probably shouldn't use it, unless you know what your doing, like you do.

Reply:
I see your point, I'll remove it